packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -632,11 +629,11 @@ export const refreshSearchIndexes = (): IndexesThunkAction<
 
     // If we are in a READY state, then we have already fetched the data
     // and are refreshing the list.
-    const newStatus: SearchIndexesStatus =
-      status === SearchIndexesStatuses.READY
-        ? SearchIndexesStatuses.REFRESHING
-        : SearchIndexesStatuses.FETCHING;
-    await dispatch(fetchIndexes(newStatus));
+    const reason: FetchReason =
+      status === FetchStatuses.READY
+        ? FetchReasons.REFRESH
+        : FetchReasons.INITIAL_FETCH;
```

Comment:
I think one thing that I was missing so far is that we're using the same action for user triggering through the refresh button, initial plugin activation, and re-fetching the indexes after we created one through the modal. I was hoping that introducing this `reason` to the action we would be able to avoid deriving the state in the action from the previous state, that is generally not recommended, but seems like specifically because this particular action is used for this initial fetching triggered from the activate, we still have to keep this around.

Using this for both user triggered and programmatic refreshes kinda makes sense to me, but do you think we can remove the initial fetch from the list of things that this action creator is used for and either have a dedicated action creator for that, or just use `fetchIndexes` with INITIAL_FETCH passed there as a reason directly in the `activate` method? I'd really want us to remove the deriving state from a previous one like that in these actions

Reply:
I made separate functions for the initial fetch, refresh and poll.